### DISCUSSION TITLE
Support for Expect: 100-continue?

### DISCUSSION THREAD

**Discussion by jamshid** [encode/httpx#1713] [Q&A] [answered]
The python `requests` library unfortunately does not support the `Expect: 100-continue` flow where the body of a PUT/POST is not sent until the server responds with a 100 Continue (https://github.com/psf/requests/issues/713).

I should try to test but does anyone know if `httpx` supports it fully? I see some tests related to it but I can't tell if it actually delays the body from being sent or if it just handles the 100 response if a server sends it (but still immediately sends the body after sending the request headers).

**Comment by lovelydinosaur** [comment, marked as answer]
Nope, we currently have the same kind of behaviour as `requests` here.
We certainly *could* support it, although I don't think it'd be a high priority item for us just yet.
